src/log.rs

```diff
@@ -425,3 +427,65 @@ where
         self.next()
     }
 }
+
+/// Returns an iterator over entries in the log stored in a thread-local variable.
+pub fn iter_thread_local<T, I, D>(
+    local_key: &'static LocalKey<RefCell<Log<T, I, D>>>,
```

Comment:
Is there a reason why we're coupling the concept of thread locals with the functionality of iteration, as opposed to adding a `.iter()` method on the `Log` structure directly as we did with `BTreeMap`.

I get that it's reasonable to assume that logs would typically be stored in thread locals, but that's not always the case. For example, in the Bitcoin canister we store stable structures inside structs and not directly as global variables ([example](https://github.com/dfinity/bitcoin-canister/blob/4b48647273f66b1832239e6dc662c317208551d0/canister/src/block_header_store.rs#L15)).

Reply:
The Log structure already has `iter()`; my problem is that I want to hide the fact that the log lives in a stable structure stored in a thread-local refcell from the rest of the code.

I want a module interface where the client can `record_event` and get an event iterator: https://sourcegraph.com/github.com/dfinity/ic@595c57c815aebc49926a91618e2cde1cb7454973/-/blob/rs/bitcoin/ckbtc/minter/src/storage.rs?L75-96

That's not possible without a custom iterator that looks up values in the thread-local.

Reply:
I see, thanks for the clarification.